frontend/src/scenes/insights/InsightTabs/RetentionTabHorizontal.tsx

```diff
@@ -0,0 +1,188 @@
+import React, { useState, useRef } from 'react'
+import { useValues, useActions } from 'kea'
+import { PropertyFilters } from 'lib/components/PropertyFilters/PropertyFilters'
+import { ActionFilterDropdown } from '../ActionFilter/ActionFilterRow/ActionFilterDropdown'
+import { entityFilterLogic } from '../ActionFilter/entityFilterLogic'
+
+import { DownOutlined, InfoCircleOutlined } from '@ant-design/icons'
+import { retentionTableLogic, dateOptions, retentionOptionDescriptions } from 'scenes/retention/retentionTableLogic'
+import { Button, Select, Tooltip, Row, Col, Skeleton } from 'antd'
+
+import { FilterType, RetentionType } from '~/types'
+import { TestAccountFilter } from '../TestAccountFilter'
+import { PropertyKeyInfo } from 'lib/components/PropertyKeyInfo'
+import './RetentionTab.scss'
+import { RETENTION_FIRST_TIME, RETENTION_RECURRING } from 'lib/constants'
+import useBreakpoint from 'antd/lib/grid/hooks/useBreakpoint'
+import { IconExternalLink } from 'lib/components/icons'
+
+export function RetentionTabHorizontal(): JSX.Element {
+    const node = useRef<HTMLElement>(null)
+    const returningNode = useRef<HTMLElement>(null)
+    const [open, setOpen] = useState<boolean>(false)
+    const [returningOpen, setReturningOpen] = useState<boolean>(false)
+    const { filters, actionsLookup, filtersLoading } = useValues(retentionTableLogic({ dashboardItemId: null }))
+    const { setFilters } = useActions(retentionTableLogic({ dashboardItemId: null }))
+
+    const screens = useBreakpoint()
+    const isSmallScreen = screens.xs || (screens.sm && !screens.md)
+
+    const entityLogic = entityFilterLogic({
+        setFilters: (filters: FilterType) => {
+            if (filters.events && filters.events.length > 0) {
+                setFilters({ target_entity: filters.events[0] })
```

Comment:
Why are we taking the 0th index here?

Reply:
This is just copied over logic from the existing component, but I think it's a port of the general insights logic for use within retention. As in, we use the first event/action as the target entity (particularly if you're switching between insights).